src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -265,3 +273,238 @@ boost::optional<GafferScene::EditScopeAlgo::TransformEdit> GafferScene::EditScop
+		if( !shader )
+		{
+			throw IECore::Exception( boost::str( boost::format( "Shader \"%1%\" does not exist" ) % parameter.shader ) );
+		}
+	}
+	else
+	{
+		shader = shaderNetwork->outputShader();
+		if( !shader )
+		{
+			throw IECore::Exception( "Output shader does not exist" );
+		}
+	}
+
+	const Data *result = shader->parametersData()->member( parameter.name );
+	if( result )
+	{
+		return result;
+	}
+	else
+	{
+		throw IECore::Exception( boost::str( boost::format( "Parameter \"%1%\" does not exist" ) % parameter.name ) );
+	}
+}
+
+} // namespace
+
+bool GafferScene::EditScopeAlgo::hasParameterEdit( const Gaffer::EditScope *scope, const ScenePlug::ScenePath &path, const std::string &attribute, const IECoreScene::ShaderNetwork::Parameter &parameter )
+{
+	return acquireParameterEdit( const_cast<EditScope *>( scope ), path, attribute, parameter, /* createIfNecessary = */ false );
+}
+
+TweakPlug *GafferScene::EditScopeAlgo::acquireParameterEdit( Gaffer::EditScope *scope, const ScenePlug::ScenePath &path, const std::string &attribute, const IECoreScene::ShaderNetwork::Parameter &parameter, bool createIfNecessary )
+{
+	string pathString;
+	ScenePlug::pathToString( path, pathString );
+
+	// If we need to create an edit, we'll need to do a compute to figure our the parameter
+	// type and value. But we don't want to do that if we already have an edit. And since the
+	// compute could error, we need to get the parameter value before making _any_ changes, so we
+	// don't leave things in a partial state. We use `ensureParameterValue()` to get the value
+	// lazily at the first point we know it will be needed.
+	ConstDataPtr parameterValue;
+	auto ensureParameterValue = [&] {
+		if( !parameterValue )
+		{
+			parameterValue = ::parameterValue( scope->outPlug<ScenePlug>(), path, attribute, parameter );
+		}
+	};
+
+	// Find processor, and row for `path`.
+
+	auto *processor = acquireParameterProcessor( scope, attribute, /* createIfNecessary = */ false );
+	if( !processor )
+	{
+		if( !createIfNecessary )
+		{
+			return nullptr;
+		}
+		else
+		{
+			ensureParameterValue();
+			processor = acquireParameterProcessor( scope, attribute, /* createIfNecessary = */ true );
+		}
+	}
+
+	auto *rows = processor->getChild<Spreadsheet::RowsPlug>( "edits" );
+	Spreadsheet::RowPlug *row = rows->row( pathString );
+	if( !row )
+	{
+		if( !createIfNecessary )
+		{
+			return nullptr;
+		}
+		ensureParameterValue();
+		row = rows->addRow();
+		row->namePlug()->setValue( pathString );
+	}
+
+	// Find cell for parameter
+
+	string tweakName = parameter.name.string();
+	if( parameter.shader.string().size() )
+	{
+		tweakName = parameter.shader.string() + "." + tweakName;
+	}
+
+	string columnName = boost::replace_all_copy( tweakName, ".", "_" );
+	if( auto *cell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( columnName ) )
+	{
+		return cell->valuePlug<TweakPlug>();
+	}
+
+	// No tweak for parameter yet. Create it.
+
```

Comment:
`createIfNecessary == false` early out?

Reply:
Thanks for catching that fine piece of muppetry. Added a check in the unit tests and fixed in 397149f.